pdcupdater/handlers/depchain/rpms.py

```diff
@@ -0,0 +1,249 @@
+""" Handlers for an RPM dependency chain model.
+
+The idea here is that these are two of what will eventually be many handlers
+that build and maintain a dependency chain in PDC.
+
+There are two handlers here.  They both respond to koji 'tag' events. One of
+them figures out what the build required at *build* time, and updates PDC with
+that. The other figures out what the build requires at *run* time, and updates
+PDC with that.
+
+"""
+
+import logging
+import time
+
+import pdcupdater.handlers
+import pdcupdater.services
+from pdcupdater.utils import (
+    rawhide_tag,
+    tag2release,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+def interesting_tags():
+    """ Returns a list of "interesting tags".
+
+    We have this hacked out for now for just rawhide.
+
+    Eventually, we should query PDC itself to figure out what tags we should be
+    concerned with.
+
+    """
+
+    #releases = bodhi_releases()
+    #stable_tags = [r['stable_tag'] for r in releases]
+    #return stable_tags + [rawhide_tag()]
+    return [rawhide_tag()]
+
+
+class BaseRPMDepChainHandler(pdcupdater.handlers.BaseHandler):
+    """ Abstract base class. """
+
+    def __init__(self, *args, **kwargs):
+        super(BaseRPMDepChainHandler, self).__init__(*args, **kwargs)
+        self.koji_url = self.config['pdcupdater.koji_url']
+
+    @property
+    def topic_suffixes(self):
+        return ['buildsys.tag']
+
+    def can_handle(self, msg):
+        if not msg['topic'].endswith('buildsys.tag'):
+            return False
+
+        # Ignore secondary arches for now
+        if msg['msg']['instance'] != 'primary':
+            log.debug("From %r.  Skipping." % (msg['msg']['instance']))
+            return False
+
+        interesting = interesting_tags()
+        tag = msg['msg']['tag']
+
+        if tag not in interesting:
+            log.debug("%r not in %r.  Skipping."  % (tag, interesting))
+            return False
+
+        return True
+
+    def _yield_pdc_relationships(self, pdc, release_id):
+        for type in self.managed_types:
```

Comment:
:+1:  I'll see if I can make it an ``abc.abstractproperty``.  If that doesn't work, I'll set it to ``None`` like you suggest.